### DISCUSSION TITLE
How do I render a specific template for the ListPart of a Content Type?

### DISCUSSION THREAD

**Discussion by DrewBrasher** [OrchardCMS/OrchardCore#10165] [Q&A] [answered]
I'm having trouble rendering a specific template for a ListPart and I have not been able to figure out what I'm doing wrong.  I have a content type called **BoardPacket** that has a **ListPart**.  I created a liquid template for the **BoardPacket** called `Content-BoardPacket.liquid` and 2 templates for the ListPart called `BoardPacket-ListPart.cshtml` and `BoardPacket-ListPart.Agenda.cshtml`.

If I put this in the `Content-BoardPacket.liquid` template it renders the template `BoardPacket-ListPart.cshtml`:
```
{{ Model.Content.ListPart | shape_render }}
```

but if I change it to this it does not work and does not render the list part at all:
```
{{ Model.Content.ListPart | shape_build_display: "Agenda" | shape_render }}
```
I based the name format on: https://docs.orchardcore.net/en/latest/docs/reference/modules/Templates/#contenttype_displaytype__parttype

I also tried `{{ Model.Content.ListPart | shape_build_display: "ListPartAgenda" | shape_render }}` with template named `BoardPacket-ListPart-ListPartAgenda.cshtml` but it still was not rendered.  I based that template name on:  https://docs.orchardcore.net/en/latest/docs/reference/modules/Templates/#contenttype_displaytype__parttype__shapetype

**Comment by ns8482e** [comment, marked as answer]
`shape_build_display` works on content item not on shape.  Here `Model.Content.ListPart` is a shape. To render different template for a shape,  Add alternate to that shape. https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#shape_add_alternates

For example, to render `BoardPacket-ListPart-Agenda.liquid` or `BoardPacket-ListPart-Agenda.cshtml` add alternate `BoardPacket__ListPart__Agenda`

E.g.

```liquid
{% shape_add_alternates Model.Content.ListPart "BoardPacket__ListPart__Agenda" %}
{{ Model.Content.ListPart | shape_render }}
```

**Reply by DrewBrasher** [reply]
Thank you!  I tried this and at first it didn't seem to work, it just rendered the `BoardPacket-ListPart.cshtml` template but I renamed that one to `BoardPacket-ListPart-Full.cshtml' and now it displays the Agenda template when I use your example.

I have another question though, can I display a shape twice on the same page using different templates?  What I eventually want is an outline of the ListPart items on the left and the full content of the list items in the main content area.  I tried this but it just displays the Agenda twice:
```
      {% shape_add_alternates Model.Content.ListPart "BoardPacket__ListPart__Agenda" %}
      {{ Model.Content.ListPart | shape_render }}
      
      {% shape_clear_alternates Model.Content.ListPart %}
      {% shape_add_alternates Model.Content.ListPart "BoardPacket__ListPart__Full" %}
      {{ Model.Content.ListPart | shape_render }}
```

**Reply by ns8482e** [reply]
You are seeing `Agenda` twice because the shape is already rendered as you are trying to render same type twice, it takes from cache 

To render same shape differently - use shape morphing by changing `Type` https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#shape_type
```liquid
      {% shape_add_alternates Model.Content.ListPart "BoardPacket__ListPart__Agenda" %}
      {{ Model.Content.ListPart | shape_render }}
      
      {% shape_clear_alternates Model.Content.ListPart %}
      {% shape_type Model.Content.ListPart "BoardPacket__ListPart__Full" %}
      {{ Model.Content.ListPart | shape_render }}
```

**Reply by DrewBrasher** [reply]
That still displays the Agenda twice.  I think I need to spend some more time learning shapes, shape types, display types, etc.  I'm still a little confused about those.  I will start a new discussion if I don't figure it out since you already answered my original question.
